src/test/java/io/gravitee/policy/basicauth/BasicAuthenticationPolicyIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+ * @author GraviteeSource Team
+ */
+@GatewayTest
+@DeployApi({ "/apis/basic-authentication.json", "/apis/basic-authentication-without-provider.json" })
+class BasicAuthenticationPolicyIntegrationTest
+    extends AbstractPolicyTest<BasicAuthenticationPolicy, BasicAuthenticationPolicyConfiguration> {
+
+    @ParameterizedTest(name = "Header value: ''{0}''")
+    @DisplayName("Should fail if no proper Authorization header for Basic authentication")
+    @ValueSource(strings = { "", "NOT_BASIC" })
+    void shouldFailIfNoProperAuthorizationHeader(String authorizationHeader, WebClient client) {
+        wiremock.stubFor(get("/endpoint").willReturn(ok()));
+
+        final TestObserver<HttpResponse<Buffer>> obs = client
+            .get("/test")
+            .putHeader(HttpHeaderNames.AUTHORIZATION, authorizationHeader)
+            .rxSend()
+            .test();
+
+        awaitTerminalEvent(obs);
+        obs
+            .assertComplete()
+            .assertValue(response -> {
+                assertThat(response.statusCode()).isEqualTo(401);
+                assertThat(response.headers().get(HttpHeaderNames.WWW_AUTHENTICATE))
+                    .isEqualTo("Basic realm=\"" + BasicAuthenticationPolicy.DEFAULT_REALM_NAME + "\"");
+                assertThat(response.bodyAsString()).isEqualTo("Unauthorized");
+                return true;
+            })
+            .assertNoErrors();
+
+        wiremock.verify(exactly(0), getRequestedFor(urlPathEqualTo("/endpoint")));
+    }
+
+    @Test
+    @DisplayName("Should fail if no resource provider configured")
+    void shouldFailIfNoProvider(WebClient client) {
+        wiremock.stubFor(get("/endpoint").willReturn(ok()));
+
+        final TestObserver<HttpResponse<Buffer>> obs = client
+            .get("/test-no-provider")
+            .putHeader(HttpHeaderNames.AUTHORIZATION, "Basic my-basic")
+            .rxSend()
+            .test();
+
+        awaitTerminalEvent(obs);
+        obs
+            .assertComplete()
+            .assertValue(response -> {
+                assertThat(response.statusCode()).isEqualTo(401);
+                assertThat(response.headers().get(HttpHeaderNames.WWW_AUTHENTICATE))
+                    .isEqualTo("Basic realm=\"" + BasicAuthenticationPolicy.DEFAULT_REALM_NAME + "\"");
+                assertThat(response.bodyAsString()).isEqualTo("No authentication provider has been provided");
+                return true;
+            })
+            .assertNoErrors();
+
+        wiremock.verify(exactly(0), getRequestedFor(urlPathEqualTo("/endpoint")));
+    }
+
+    @Test
+    @DisplayName("Should fail if token is not base64")
+    void shouldFailIfTokenNotBase64(WebClient client) {
+        wiremock.stubFor(get("/endpoint").willReturn(ok()));
+
+        final TestObserver<HttpResponse<Buffer>> obs = client
+            .get("/test")
+            .putHeader(HttpHeaderNames.AUTHORIZATION, "Basic my-basic")
```

Comment:
just wondering if we should not use a valid username here in order to fully match the test descriptor, WDYT ?